File: cs/src/AutoDiff/Compiled/ITapeVisitor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace AutoDiff.Compiled
{
    interface ITapeVisitor
    {
        void Visit(Constant elem);
        void Visit(Exp elem);
        void Visit(Log elem);
        void Visit(ConstPower elem);
        void Visit(TermPower elem);
        void Visit(Product elem);
        void Visit(Sum elem);
		void Visit(Gp elem);
        void Visit(Variable var);
		void Visit(Min elem);
        void Visit(Max elem);
        void Visit(And elem);
        void Visit(Or elem);            
        void Visit(Sigmoid elem);
		void Visit(LinSigmoid elem);
        void Visit(LTConstraint elem);
        void Visit(LTEConstraint elem);
        void Visit(ConstraintUtility elem);
        void Visit(Sin elem);
        void Visit(Cos elem);
        void Visit(Abs elem);
        void Visit(Atan2 elem);
        void Visit(Reification elem);        
    }
}

File: java/src/main/java/AutoDiff/Compiled/ITapeVisitor.java
//
// Translated by CS2J (http://www.cs2j.com): 12.11.2016 09:40:32
//

package AutoDiff.Compiled;

public interface ITapeVisitor
{
    void visit(Constant elem) throws Exception ;

    void visit(Exp elem) throws Exception ;

    void visit(Log elem) throws Exception ;

    void visit(ConstPower elem) throws Exception ;

    void visit(TermPower elem) throws Exception ;

    void visit(Product elem) throws Exception ;

    void visit(Sum elem) throws Exception ;

    void visit(Variable var) throws Exception ;

    void visit(Min elem) throws Exception ;

    void visit(Max elem) throws Exception ;

    void visit(And elem) throws Exception ;

    void visit(Or elem) throws Exception ;

    void visit(Sigmoid elem) throws Exception ;

    void visit(LinSigmoid elem) throws Exception ;

    void visit(LTConstraint elem) throws Exception ;

    void visit(LTEConstraint elem) throws Exception ;

    void visit(ConstraintUtility elem) throws Exception ;

    void visit(Sin elem) throws Exception ;

    void visit(Cos elem) throws Exception ;

    void visit(Abs elem) throws Exception ;

    void visit(Atan2 elem) throws Exception ;

    void visit(Reification elem) throws Exception ;

}


